external-crates/move/crates/move-compiler/tests/linter/move_2024/unneeded_return_okay.exp

```diff
@@ -0,0 +1,54 @@
+warning[W09002]: unused variable
+   ┌─ tests/linter/move_2024/unneeded_return_okay.move:12:9
+   │
+12 │     let y = &mut x;
+   │         ^ Unused local variable 'y'. Consider removing or prefixing with an underscore: '_y'
+   │
+   = This warning can be suppressed with '#[allow(unused_variable)]' applied to the 'module' or module member ('const', 'fun', or 'struct')
+
+error[E04007]: incompatible types
+   ┌─ tests/linter/move_2024/unneeded_return_okay.move:13:14
+   │
+11 │     let mut x = 0;
+   │         ----- Given: integer
+12 │     let y = &mut x;
+13 │     return (*x = 1)
+   │              ^
+   │              │
+   │              Invalid mutation. Expected a mutable reference
+   │              Expected: '&mut _'
+
+error[E04007]: incompatible types
+   ┌─ tests/linter/move_2024/unneeded_return_okay.move:16:12
+   │
+16 │ fun t3() { 'a: { return { return 'a 0 } } }
+   │     --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │     │      │
+   │     │      Invalid return expression
+   │     │      Given: integer
+   │     Expected: '()'
```

Comment:
Are these intentional?

Reply:
Yes, they are error states that mean the lint doesn't run.